### DISCUSSION TITLE
Fire Simulation Modeling for muttiple burning materials

### DISCUSSION THREAD

**Discussion by MohamedSalah989** [firemodels/fds#12744] [Q&A]
Hello everyone,

I am working on a fire simulation scenario involving multiple materials for a facade system with a product of combustion for each material and different heat of combustion and ignition temperatures.

Can i sum the product of combustion of the two material as the pyrosim only accept one active reaction.
or how can I identify the reaction for each and both materials?

Also, how can i get the HRR curve due to combustion for each material and how can i get the accumulated HRR for both materials?

finally , what is the effect of using the option of allow obstructions to burn away or not.

I would greatly appreciate your support.

![FDS case](https://github.com/firemodels/fds/assets/138606386/bef1f00f-f3f4-4596-9246-92e00ef5ec9c)

Best regards,
Mohamed Salah
PhD Student

**Comment by Er9y714** [comment]
Pyrosim is not offering the full set of features of FDS. You will only find the most frequently used settings. Therefore I recommend reading the FDS User Guide and understand how to setup the model in FDS codes. Then if possible, you can model in Pyrosim. For such complex cases I would recommend double checking the fds file that Pyrosim outputs. 

If the setup is not possible to create on Pyrosim, writing the FDS file manually is the only solution. FDS User Guide and verification examples should be possible to do what you aim. See [User guide and Verification guide ](https://pages.nist.gov/fds-smv/manuals.html)  and [verification example files.](https://github.com/firemodels/fds/tree/master/Verification) 

You can also find the necessary information of burn away feature in the user guide.

**Reply by MohamedSalah989** [reply]
@Er9y714 , many thanks for your reply, i will search more in the FDS user guidelines.

Regards
Mohamed Salah

**Comment by MohamedSalah989** [comment]
@Er9y714 

I tried to create FDS file to contain the thermal properties, product of combustion, and  ignition of temperatures for different material, but i faced the below error: Could you advise me on the right action to solve this issue?

![FDS Case](https://github.com/firemodels/fds/assets/138606386/f7324fc6-c1c3-465c-840d-9e38c59b8f36)

**Reply by Er9y714** [reply]
Search the error in the user guide. it will lead you to the relevant section 9.1.3

**Comment by MohamedSalah989** [comment]
@Er9y714 
the error show that i can't identify the ignition temperature and the thermal properties for a material and i search in the FDS guideline but i didn't find a clear answer for it. my CFD case assign that the material shall be start ignition at certain temperature with the other thermal properties. and i cant know how to achieve that.

**Comment by Er9y714** [comment]
Can you share the example simplified fds file?

**Reply by MohamedSalah989** [reply]
@Er9y714 
Sure, please find attached FDS file.
[C3.zip](https://github.com/firemodels/fds/files/14923947/C3.zip)

**Reply by Er9y714** [reply]
The first thing I notice is that the OBSTructions do not align with the mesh. So when FDS runs the input, the input is snapping to the nearest cells. But it is not clear if the processed geometry matches with the intended geometry. I suggest you to fix the geometry as a first step. Make sure the obstructions fill the cells (zero thickness obstruction is allowed but best to avoid to prevent specific case issues). Then report back if you still get the same error. 

**Requested**
![image](https://github.com/firemodels/fds/assets/26166281/f8c03742-2943-4e57-a5df-36ac9d29b44a)

**FDS interpreted**
![image](https://github.com/firemodels/fds/assets/26166281/6427a89f-cad4-4a5a-a65b-f310d6f55a22)

**Reply by MohamedSalah989** [reply]
@Er9y714 
Many thanks for your answer; i will follow that and give feedback to you, highly appreciated!

**Reply by MohamedSalah989** [reply]
@Er9y714 
The issue is identifying the combustion product in the material section and identifying the ignition temperature in the surface section, When the product of combustion is removed, the simulation is running well.
my question is: how do I identify the product of combustion for each material if i cannot put them in the material section?
